### ISSUE TITLE
remove all mentions of plugins.tml from the documentation

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3246]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)| @radix reported|
|-|-|
|Trac ID|trac#3246|
|Type|task|
|Created|2008-05-20 23:50:37Z|
|Branch|https://github.com/twisted/twisted/tree/remove-plugins-tml-3246-2|

There are a couple mentions in The Tutorial, and one in the TwistedQuotes document:

```
./howto/quotes.xhtml:    <li><a href="listings/TwistedQuotes/plugins.tml" class="py-listing">plugins.tml</a>.</li>
./howto/tutorial/configuration.xhtml:<a href="listings/finger/finger/plugins.tml" class="py-listing">finger/plugins.tml</a>
./howto/tutorial/listings/finger/finger/plugins.tml:#finger/plugins.tml
```

<details><summary>Searchable metadata</summary>

```
trac-id__3246 3246
type__task task
reporter__radix radix
priority__normal normal
milestone__ 
branch__branches_remove_plugins_tml_3246_2 branches/remove-plugins-tml-3246-2
branch_author__thijs__exarkun thijs, exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1211327437000000 1211327437000000
changetime__1229359127000000 1229359127000000
version__None None
owner__ 
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_1"></a>|@radix commented|
|-|-|

There's also a few in the lore documentation.

**Comment by twisted-trac** [comment]
|<img alt="collab's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|collab set owner to collab<br>collab set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra set owner to @thijstriemstra<br>@thijstriemstra set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_5"></a>|@thijstriemstra commented|
|-|-|

(In [24528]) Branching to 'remove-plugins-tml-3246'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_6"></a>|@thijstriemstra set status to `assigned`|
|-|-|

I assume the task here is to rewrite the .tml examples/documentation and convert them to the new plugin system as described on [this](http://twistedmatrix.com/projects/core/documentation/howto/plugin.html) page.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_7"></a>|@thijstriemstra removed owner<br>@thijstriemstra set status to `new`|
|-|-|

I'm not very familiar with the plugins system in Twisted but read the docs and attempted to move the TwistedQuotes app to the twisted.plugin.IPlugin approach. The TwistedQuotes page doesn't describe how to actually run the sample without the use of mktap, if I understand correctly, so if someone can let me know if I'm 1) on the right path with the plugin approach in r24530 and 2) what the preferred approach of running a plugin with twistd/.tac's is.

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_8"></a>|@radix set owner to @thijstriemstra|
|-|-|

Ok, so I don't think that TwistedQuotes should have plugins at all. None of the documents that refer to it actually talk about plugins (except for a misuse in design.xhtml, which refers to TwistedQuotes itself as a "plugin" but should instead use the word "package"). For example, none of the documents actually explain what that 'plugins.tml' file was doing in the TwistedQuotes package; they were probably removed a while ago.

I think we already have a good enough example of plugins in the plugins document itself, and including plugins support in TwistedQuotes is likely to just confuse readers.

So I think you should just get rid of the plugins.tml file, get rid of the reference to it in quotes.xhtml, and update design.xhtml to refer to TwistedQuotes as a "package" instead of a "plugin".

Sorry for the long review period, thijs.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_9"></a>|@thijstriemstra removed owner|
|-|-|

Replying to [radix](#note_8):
> So I think you should just get rid of the plugins.tml file, get rid of the reference to it in quotes.xhtml, and update design.xhtml to refer to TwistedQuotes as a "package" instead of a "plugin".

Made the suggested changes in r24620, putting it up for review.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_10"></a>|@exarkun set owner to @thijstriemstra|
|-|-|

  1. `doc/core/howto/listings/TwistedQuotes/quoteproto.py` only has whitespace changes, and they look like they change the code to not conform to the coding standard.
  1. Ditto for `doc/core/howto/listings/TwistedQuotes/quoters.py`
  1. There are some more mentions of plugins.tml:
    * `doc/lore/howto/extend-lore.xhtml`
    * `doc/lore/howto/listings/lore/plugins.tml`
    * `doc/core/howto/tutorial/configuration.xhtml`
    * `doc/core/howto/tutorial/listings/finger/finger/plugins.tml`

Let me know if you'd still like an answer to the questions you asked in your comment prior to radix's review.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_11"></a>|@thijstriemstra removed owner|
|-|-|

Replying to [exarkun](#note_10):
>   1. `doc/core/howto/listings/TwistedQuotes/quoteproto.py` only has whitespace changes, and they look like they change the code to not conform to the coding standard.
>   1. Ditto for `doc/core/howto/listings/TwistedQuotes/quoters.py`

Right, reverted that change in r24641

>   1. There are some more mentions of plugins.tml:
>     * `doc/lore/howto/extend-lore.xhtml`
>     * `doc/lore/howto/listings/lore/plugins.tml`
I can't just rip this lore example out without influencing the rest of the howto, or..? What to do with this?

>     * `doc/core/howto/tutorial/configuration.xhtml`
>     * `doc/core/howto/tutorial/listings/finger/finger/plugins.tml`

Should this tml/mktap stuff be replaced by a .tac?

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_12"></a>|@exarkun set owner to @thijstriemstra|
|-|-|

> I can't just rip this lore example out without influencing the rest of the howto, or..? What to do with this? 

Lore supports new-style plugins too, so I think the change we need to make is to convert the .tml file to a new-style plugin and update the prose which talks about it to agree.  Hopefully a lot of text will be obsolete or can be replaced with a reference to the plugin howto.

> Should this tml/mktap stuff be replaced by a .tac? 

A twistd plugin probably makes more sense and is a smaller change.  All mktap plugins are twistd plugins by default.  So this would become like "twistd finger ..." instead of "mktap finger ...; twistd -f finger.tap".

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_13"></a>|@exarkun commented|
|-|-|

(In [25643]) replace the old-style lore plugin with a new-style lore plugin and update the documentation

refs [#3246](https://github.com/twisted/twisted/issues/3246)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_14"></a>|@exarkun commented|
|-|-|

(In [25644]) replace the old-style finger plugin with a new-style twistd plugin and update the documentation

refs [#3246](https://github.com/twisted/twisted/issues/3246)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_15"></a>|@exarkun commented|
|-|-|

(In [25645]) stop talking about TwistedQuotes entirely

The plugin stuff is all irrelevant to this pb document.  The features relevant
to pb explained by TwistedQuotes are already explained in other parts of this
document.

refs [#3246](https://github.com/twisted/twisted/issues/3246)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_16"></a>|@exarkun commented|
|-|-|

(In [25646]) get rid of another TwistedQuotes tml file

refs [#3246](https://github.com/twisted/twisted/issues/3246)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_17"></a>|@exarkun commented|
|-|-|

(In [25647]) Branching to 'remove-plugins-tml-3246-2'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_18"></a>|@exarkun removed owner|
|-|-|

Fixed the remaining issues, I think.  Please review.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_19"></a>|@thijstriemstra set owner to @exarkun|
|-|-|

Looks good to me, not sure why you're commenting out some stuff in extend-lore.xhtml though?

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_20"></a>|@exarkun commented|
|-|-|

I commented out the stuff in lore because it wasn't providing any value to readers.  It might be useful to a future lore maintainer as a hint about what documentation still needs to be written, though.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_21"></a>|@exarkun set status to `closed`|
|-|-|

(In [25672]) Merge remove-plugins-tml-3246-2

Author: thijs, exarkun, itamarst
Reviewer: radix, thijs
Fixes: [#3246](https://github.com/twisted/twisted/issues/3246)

Remove from the documentation all remaining references to the old style of plugins
and replace them with documentation based on the new style of plugins, where
appropriate.  The Lore plugin interface `IProcess` is also slightly improved so
that it at least defines the attributes which are actually required of it and
does not document an attribute which is never used by Lore.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
